Tests/DataPipeline/Support/XCTestCaseExtensions.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import XCTest
+
+extension XCTest {
+    /// Asserts that the actual dictionary matches the expected dictionary, with type-specific comparisons. Assumes `String` as the default type for values unless explicitly specified.
+    func XCTAssertMatches(
```

Comment:
This method has helped me reduce boilerplate code in dictionary comparisons

Reply:
Could you give some examples of code that uses this, showing the reason why it's needed? 

I feel like something like this should already exist in the XCTest framework? So I wanted to ask this question to see. I would prefer we can leverage something already in the framework then maintain this function in the long-term.

Reply:
If we do decide to keep this, I suggest moving it into `SharedTest` module so all test modules can use it.

Reply:
> Could you give some examples of code that uses this

I've utilized it for checking user traits and other event properties. You can its usage in [`DataPipelineInteractionTests`](https://github.com/customerio/customerio-ios/blob/rehan/cdp-test-base/Tests/DataPipeline/DataPipelineInteractionTests.swift) and [`DataPipelineCompatibilityTests`](https://github.com/customerio/customerio-ios/blob/rehan/cdp-json-compat-tests/Tests/DataPipeline/DataPipelineCompatibilityTests.swift).

> I feel like something like this should already exist in the XCTest framework

I couldn't find a similar method in the XCTest framework. It may be likely due to `Any` not being `Equatable`, making direct comparison of `[String : Any]` challenging. I've written this method to so we can transform `Any` into some `Equatable` and compare.

> I suggest moving it into SharedTest module so all test modules can use it

I decided not to include it in `SharedTest` as its current use seemed limited outside the `DataPipeline` module. But I'm open to relocating it if it's more beneficial or easier to maintain there.

Reply:
> I couldn't find a similar method in the XCTest framework. It may be likely due to Any not being Equatable, making direct comparison of [String : Any] challenging

Ah, that could be. I can see value in a convenient [String: Any] equal assertion function. 

If the goal is checking equality of 2 `[String: Any]` objects, I do think there could be a more simple and flexible solution. One that is also easier to maintain in the future. 

I checked some of the exiting custom attribute tests in the code base since those existing tests would also be checking `[String: Any]` equality. 

There might be another strategy being used in the code base today, but [what I see thus far](https://github.com/customerio/customerio-ios/blob/10d9f6cde37af5001160b0a9af853618ac3440e3/Tests/Tracking/CustomerIOImplementationTest.swift#L61) is to create a JSON string from the `[String: Any]` data type and then do a string comparison of the 2 strings. 

Do you think this could work for your use case?

Reply:
Unfortunately this seems to work well for `[String: String]` only and not `[String: Any]`

Merging this for now, but keep sharing ideas so we can improve it and opt for simpler solution if possible.